### PR TITLE
APPDEV-1161 Enable app release to production

### PR DESCRIPTION
**Pull request by Bert-R**
* Added the required stages
* Uncommented the lane

Along with this:
* Move the logic for incrementing the version number into a function
* Extended the commit message with the version code
* Switched from "deploy" and "promote" to "release", in alignment with the terminology on Google Play

### CODE REVIEW

**Comment by Sivateja-CMI** [comment]
I am fine with the changes. @aristotelos please review and merge.

File: Jenkinsfile [hunk 1]
```diff
@@ -106,15 +95,15 @@ pipeline {
...
         script {
-          env.DEPLOY_AS_BETA = input message: 'User input required',
+          env.RELEASE_TO_BETA = input message: 'User input required',
               submitter: 'authenticated',
-              parameters: [choice(name: 'Promote to beta on Google Play', choices: 'no\nyes', description: 'Choose "yes" if you want to promote this to beta on Google Play')]
+              parameters: [choice(name: 'Release to beta on Google Play', choices: 'no\nyes', description: 'Choose "yes" if you want to release this build to beta on Google Play')]
         }
       }
     }
     stage('Promote to beta') {
```
Comment (aristotelos) [inline Jenkinsfile]: 'Promote to beta' => 'Release to beta'

File: Jenkinsfile [hunk 2]
```diff
@@ -124,12 +113,62 @@ pipeline {
       }
       post {
         success {
-          slackSend color: 'good', channel: '#dev', message: "Android app build ${env.BUILD_NUMBER} on branch ${BRANCH_NAME} successfully promoted to beta on Google Play"
+          slackSend color: 'good', channel: '#dev', message: "Android app build ${env.BUILD_NUMBER} on branch ${BRANCH_NAME} successfully released to beta on Google Play"
+        }
+        failure {
+          slackSend color: 'bad', channel: '#dev', message: "Android app build ${env.BUILD_NUMBER} on branch ${BRANCH_NAME} failed to released to beta on Google Play"
```
Comment (aristotelos) [inline Jenkinsfile]: `failed to released` => `failed to release`

File: Jenkinsfile [hunk 3]
```diff
@@ -124,12 +113,62 @@ pipeline {
...
+      steps {
+        checkpoint 'App released to beta'
+        script {
+          env.RELEASE_TO_PRODUCTION = input message: 'User input required',
+              submitter: 'authenticated',
+              parameters: [choice(name: 'Release to production on Google Play', choices: 'no\nyes', description: 'Choose "yes" if you want to release this build to production on Google Play')]
+        }
+      }
+    }
+    stage('Promote to production') {
```
Comment (aristotelos) [inline Jenkinsfile]: 'Release to production'

File: Jenkinsfile [hunk 4]
```diff
@@ -124,12 +113,62 @@ pipeline {
...
+      }
+      steps {
+        sh 'cd app && bundle install'
+        withCredentials(bindings: [string(credentialsId: 'GoogleJsonKeyData', variable: 'SUPPLY_JSON_KEY_DATA')]) {
+          sh 'cd app && bundle exec fastlane --verbose production'
+        }
+      }
+      post {
+        success {
+          slackSend color: 'good', channel: '#dev', message: "Android app build ${env.BUILD_NUMBER} on branch ${BRANCH_NAME} successfully release to production on Google Play"
```
Comment (aristotelos) [inline Jenkinsfile]: 'successfully release' => 'successfully released'

**Pull request merged**